src/scores/probability/rev_impl.py

```diff
@@ -0,0 +1,860 @@
+    # False alarm rate (probability of false detection) equation 5
+
+    try:
+        check_monotonic_array(cost_loss_ratios)
+    except Exception as ex:
+        raise type(ex)("for cost_loss_ratios, " + str(ex))
+
+    if cost_loss_dim in (set(pod.dims) | set(pofd.dims) | set(climatology.dims)):
+        raise ValueError(f"dimension '{cost_loss_dim}' must not be in input data")
+
+    # Handle Dataset inputs by applying to each variable
+    if isinstance(pod, xr.Dataset):
+        result_dict = {}
+        for var in pod.data_vars:
+            result_dict[var] = relative_economic_value_from_rates(
+                pod[var],
+                pofd[var] if isinstance(pofd, xr.Dataset) else pofd,
+                climatology[var] if isinstance(climatology, xr.Dataset) else climatology,
+                cost_loss_ratios,
+                cost_loss_dim=cost_loss_dim,
+            )
+        return xr.Dataset(result_dict)
+
+    alphas = xr.DataArray(
+        cost_loss_ratios,
+        dims=[cost_loss_dim],
+        coords={cost_loss_dim: cost_loss_ratios},
+    )
+
+    obar, alphas = xr.broadcast(climatology, alphas)
+    climatological_term = xr.where(obar < alphas, obar, alphas)
+
+    # calculate the relative economic value (equation 8)
+
+    rev = ((climatological_term) - (pofd * alphas * (1 - obar)) + (pod * obar * (1 - alphas)) - obar) / (
+        climatological_term - obar * alphas
+    )
+
+    # tidy up floating point infinities - necessary because you can get
+    # near-zero in the denominator for alpha=0 or alpha=1
+    rev = rev.where(~np.isinf(rev))
+
+    return rev
+
+
+def _create_output_dataset(
+    rev: xr.DataArray,
+    thresholds: Sequence[float],
+    cost_loss_ratios: Sequence[float],
+    derived_metrics: Optional[Sequence[str]],
+    threshold_outputs: Optional[Sequence[float]],
+    threshold_dim: str,
+    cost_loss_dim: str,
+) -> xr.Dataset:
+    """
+    Create output Dataset with derived metrics and threshold slices.
+
+    Args:
+        rev: Full REV DataArray with threshold and cost_loss_ratio dimensions
+        thresholds: List of threshold values
+        cost_loss_ratios: List of cost-loss ratio values
+        derived_metrics: Derived metrics to compute
+        threshold_outputs: Specific thresholds to extract
+        threshold_dim: Name of threshold dimension
+        cost_loss_dim: Name of cost-loss ratio dimension
+
+    Returns:
+        Dataset with requested outputs
+
+    Raises:
+        ValueError: If invalid derived_metrics values provided
+        ValueError: If 'rational_user' requested but thresholds don't match cost_loss_ratios
+    """
+    derived_metrics = [] if derived_metrics is None else list(derived_metrics)
+    threshold_outputs = [] if threshold_outputs is None else list(threshold_outputs)
+
+    # Store original attributes
+    attributes = rev.attrs
+    rev.attrs = {}
+
+    result = xr.Dataset()
+    result.attrs = attributes
```

Comment:
```suggestion
    result = xr.Dataset(attrs=rev.attrs)
```
Can we simplify it like this?

Reply:
Sure, done.